ani-cli

```diff
@@ -193,15 +193,19 @@ get_episode_url() {
     links=$(cat "$cache_dir"/* | sed 's|^Mp4-||g;/http/!d;/Alt/d' | sort -g -r -s)
     rm -r "$cache_dir"
     episode=$(select_quality "$quality")
-    [ -z "$episode" ] && die "Episode not released!"
+    if printf "%s" "$ep_list" | grep -q "^$ep_no$"; then
+        [ -z "$episode" ] && die "Episode is released, but no valid sources!"
+    else
+        [ -z "$episode" ] && die "Episode not released!"
+    fi
 }
 
 # search the query and give results
 search_anime() {
     search_gql="query(        \$search: SearchInput        \$limit: Int        \$page: Int        \$translationType: VaildTranslationTypeEnumType        \$countryOrigin: VaildCountryOriginEnumType    ) {    shows(        search: \$search        limit: \$limit        page: \$page        translationType: \$translationType        countryOrigin: \$countryOrigin    ) {        edges {            _id name availableEpisodes __typename       }    }}"
 
     curl -e "$allanime_refr" -s -G "${allanime_api}/api" --data-urlencode "variables={\"search\":{\"allowAdult\":false,\"allowUnknown\":false,\"query\":\"$1\"},\"limit\":40,\"page\":1,\"translationType\":\"$mode\",\"countryOrigin\":\"ALL\"}" --data-urlencode "query=$search_gql" -A "$agent" | sed 's|Show|\
-|g' | sed -nE "s|.*_id\":\"([^\"]*)\",\"name\":\"([^\"]*)\".*${mode}\":([1-9][^,]*).*|\1	\2 (\3 episodes)|p"
+| g' | sed -nE "s|.*_id\":\"([^\"]*)\",\"name\":\"(.+)\",.*${mode}\":([1-9][^,]*).*|\1\t\2 (\3 episodes)|p" | sed 's/\\"//g'
```

Comment:
The literal tab may be necessary on bsd systems, those have trouble with whitespace regex characters, which is why we have some ugly literal newlines in a few places

Reply:
I have a commit ready to go to fix that. I searched for other \t-s, and see quite a few. Most are within `printf` statements, which it looks like supports the \t. I did see one in a `tr` (25) and a few within another `sed` (244). I also see \t mentioned in the `sed` docs for bsd, but I didn't quite get the context.
I don't know if those are worth getting into though in this PR.
I can just commit the literal tab for this particular instance and call it there.

Reply:
Sed on 244 is fishy too then, but the other coreutils should be fine.
It was pretty much only sed that gave us issues

Reply:
I don't want to overstep, but i can push a commit removing the ones on 244.  Pretty quick fix.

Reply:
Sorry for the late reply. Yes, please do. Otherwise this PR is about ready

Reply:
No worries! Pushed that commit and I believe it's RTG!